### PR TITLE
Hmac separate

### PR DESCRIPTION
**Pull request by sosthene-nitrokey**
Built on top of #15 

This PR stops depending on trussed-staging for the never-merged HMACSHA256P256 extension and uses a custom backend instead (https://github.com/Nitrokey/trussed-hmacsha256p256)

### CODE REVIEW

**Comment by robin-nitrokey** [comment]
As the extension is only intended to be used with this crate as a workaround, we could also put it directly into this repository.  That would probably be easier to maintain.

**Comment by sosthene-nitrokey** [comment]
Right, that's probably a better idea indeed.

**Comment by sosthene-nitrokey** [comment]
I moved the backend here.

File: Cargo.toml [hunk 1]
```diff
@@ -63,11 +66,11 @@ apdu-peek = []
 
 # Support P256 key derivation from HMAC
 # Uses trussed-staging
```
Comment (robin-nitrokey) [inline Cargo.toml]: outdated comment

File: Cargo.toml [hunk 2]
```diff
@@ -63,11 +66,11 @@ apdu-peek = []
...
-hmacsha256p256 = ["trussed-staging/hmacsha256p256"]
+hmacsha256p256 = ["hmac", "sha2"]
 
 # Allow to inject raw key data with specified type into the Trussed key store. Needed for OpenPGP support to use Kind::P256.
 # Uses trussed-staging
```
Comment (robin-nitrokey) [inline Cargo.toml]: outdated comment

File: Cargo.toml [hunk 3]
```diff
@@ -109,7 +112,8 @@ trussed-auth = { git = "https://github.com/trussed-dev/trussed-auth", rev = "622
 trussed-hkdf = { git = "https://github.com/Nitrokey/trussed-hkdf-backend.git", tag = "v0.1.0" }
 trussed-rsa-alloc = { git = "https://github.com/trussed-dev/trussed-rsa-backend.git", rev = "2f51478f0861ff8db19fdd5290f023ab6f4c2fb9" }
 trussed-usbip = { git = "https://github.com/Nitrokey/pc-usbip-runner", tag = "v0.0.1-nitrokey.1" }
-trussed-staging = { git = "https://github.com/Nitrokey/trussed-staging.git", branch = "hmacsha256p256-chunked" }
+trussed-staging = { git = "https://github.com/Nitrokey/trussed-staging.git", rev = "6e70fc4db31b92c413d56e28c6d3683a87384d46" }
+trussed-hmacsha256p256 = { git = "https://github.com/Nitrokey/trussed-hmacsha256p256.git", rev = "9a12feb358437628844b844cddd08e2de0c36b3c" }
```
Comment (robin-nitrokey) [inline Cargo.toml]: unnecessary patch

**Review by robin-nitrokey — COMMENTED** [review]
LGTM, just some outdated lines in the `Cargo.toml`.

**Comment by sosthene-nitrokey** [comment]
Should we tag with this PR or with main also?

**Review by robin-nitrokey — APPROVED** [review]

**Review by daringer — APPROVED** [review]

**Pull request merged**